src/main/java/com/avispl/symphony/dal/device/pjlink/PJLinkCommunicator.java

```diff
@@ -1097,11 +1093,13 @@ private void validateControllableProperty(String commandResponseValue, String na
                 }
                 throw new IllegalArgumentException("Unsupported control command: " + name);
             case PJLinkConstants.OUT_OF_PARAMETER:
-                throw new IllegalArgumentException("Missing control command parameter: " + name);
+                throw new IllegalArgumentException("Missing or incorrect control command parameter: " + name);
             case PJLinkConstants.UNAVAILABLE_TIME:
                 throw new IllegalStateException("Unable to send control command due to the device state");
             case PJLinkConstants.DEVICE_FAILURE:
                 throw new RuntimeException("Unable to send control command due to the general device failure");

```

Comment:
throwing a RuntimeException can lead to subtle errors; for example, a caller cannot examine the exception to determine why it was thrown and consequently cannot attempt recovery. Could change this to another exception type.

Reply:
Yeah, generally it is one of such exceptions - there's not enough information on why operation fails. 
The second one is changed to FailedLoginException though :)